packages/web-components/src/components/cta-section/cta-section.ts

```diff
@@ -35,20 +37,44 @@ const slotExistencePropertyNames = {
  */
 @customElement(`${ddsPrefix}-cta-section`)
 class DDSCTASection extends StableSelectorMixin(DDSContentItem) {
+  /**
+   * Content Items nodes.
+   */
+  @query(`slot[name='content-item']`)
+  private _contentsNode?: HTMLSlotElement;
+
   /**
    * `true` if there is CTA content.
    */
   @internalProperty()
   protected _hasFooter = false;
 
   /**
-   * Applies section attribute
+   * The observer for the resize of the content item slot
    */
-  connectedCallback() {
-    if (!this.hasAttribute('role')) {
-      this.setAttribute('role', 'section');
+  private _observerResizeContainer: any | null = null; // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+
+  /**
+   * Cleans-up and creats the resize observer for the content-item slot
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new resize observer.
+   */
+  private _cleanAndCreateObserverResize({ create }: { create?: boolean } = {}) {
+    const { _contentsNode: contentsNode } = this;
+
+    if (contentsNode?.assignedNodes().length) {
+      if (this._observerResizeContainer) {
+        this._observerResizeContainer.disconnect();
+        this._observerResizeContainer = null;
+      }
+
+      if (create) {
+        setTimeout(() => {
+          this._observeResizeContainer();
+        });
```

Comment:
`this._observeResizeContainer()` shouldn't be called directly, because `ResizeObserver` should determine the most optimal timing to call `this._observeResizeContainer()`. If you don't see `this._observeResizeContainer()` invoked from `ResizeObserver` in below way please let me know.

```suggestion
        // TODO: Wait for `.d.ts` update to support `ResizeObserver`
        // @ts-ignore
        this._observerResizeContainer = new ResizeObserver(this._observeResizeContainer);
        this._observerResizeContainer.observe(this._contentsNode);
```

Reply:
That's good to know, in the latest commit I added this in, as well as refactored the actual height-changing code.